python/src/equistore/operations/equal_metadata.py

```diff
@@ -0,0 +1,169 @@
+"""
+Module for checking equivalence in metadata between 2 TensorMaps
+"""
+from typing import List, Optional
+
+from ..tensor import TensorMap
+from ._utils import _check_blocks, _check_maps, _check_same_gradients
```

Comment:
I think it makes sense to move these function in this file, since they are very related. 

It might also be worth to make all functions use `equal_metadata` instead of `_check_XXX`, but this can be a future PR.